src/jit/x64/emit.dasc

```diff
@@ -1917,9 +1917,38 @@ static void load_call_arg(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJit
         | mov TMP6, qword WORK[arg.v.reg];
         | lea TMP6, STOOGE:TMP6->data;
         break;
+    case MVM_JIT_REG_DYNIDX:
+        | get_cur_op TMP5;
+        | xor TMP6, TMP6;
+        | mov TMP6w, word [TMP5 + arg.v.reg*2]
+        | mov TMP6, qword [WORK + TMP6*8];
+        break;
     case MVM_JIT_DATA_LABEL:
         | lea TMP6, [=>(arg.v.lit_i64)];
         break;
+    case MVM_JIT_SAVED_RV:
+        | mov TMP6, qword [rbp-0x28];
+        break;
+    case MVM_JIT_ARG_I64:
+        | mov TMP6, TC->cur_frame;
+        | mov TMP6, FRAME:TMP6->args;
+        | mov TMP6, qword REGISTER:TMP6[arg.v.lit_i64];
+        break;
```

Comment:
The frame args and frame local variables are always contiguous (or used to be, maybe @jnthn knows), so *if* you know the size of the whole frame you can compute the offset directly from  args (and save yourself having to read FRAME->args, because you can just use rbx instead).

Reply:
Yes, it would be safe to do that.

Reply:
At the time when this code is compiled, the code calling ncinvoke_o does not even exist yet. So while args is always part of WORK, I don't know yet where exactly in WORK it will end up in. Hence the need to read FRAME->args.

Later when the frame is speshed and compiled again, this information is available and I emit MVM_JIT_PARAM_I64 instead of MVM_JIT_ARG_I64. The former can access WORK directly, as the offsets are known.